src/test/scala/org/constellation/domain/redownload/RedownloadServiceTest.scala

```diff
@@ -1,72 +1,231 @@
+      val redownloadService = RedownloadService[IO](dao)
+      val firstSnapshot = RecentSnapshot("aaaa", 2L, Map.empty)
+      val secondSnapshot = RecentSnapshot("bbbb", 4L, Map.empty)
+      val persistFirst = redownloadService.persistOwnSnapshot(2L, firstSnapshot)
+      val persistSecond = redownloadService.persistOwnSnapshot(4L, secondSnapshot)
       val check = redownloadService.getOwnSnapshots()
 
-      (persistFirst >> persistSecond >> check).unsafeRunSync shouldBe Map(2L -> "aaaa", 4L -> "bbbb")
+      (persistFirst >> persistSecond >> check).unsafeRunSync shouldBe Map(2L -> RecentSnapshot("aaaa", 2L, Map.empty),
+                                                                          4L -> RecentSnapshot("bbbb", 4L, Map.empty))
     }
   }
 
   "getOwnSnapshot" - {
     "should return hash if snapshot at given height exists" in {
-      val redownloadService = RedownloadService[IO]()
+      val redownloadService = RedownloadService[IO](dao)
+      val newSnapshot = RecentSnapshot("aaaa", 2L, Map.empty)
 
-      val persist = redownloadService.persistOwnSnapshot(2L, "aaaa")
+      val persist = redownloadService.persistOwnSnapshot(2L, newSnapshot)
       val check = redownloadService.getOwnSnapshot(2L)
 
-      (persist >> check).unsafeRunSync shouldBe "aaaa".some
+      (persist >> check).unsafeRunSync shouldBe RecentSnapshot("aaaa", 2L, Map.empty).some
     }
 
     "should return None if snapshot at given height does not exist" in {
-      val redownloadService = RedownloadService[IO]()
+      val redownloadService = RedownloadService[IO](dao)
 
       val check = redownloadService.getOwnSnapshot(2L)
 
       check.unsafeRunSync shouldBe none[String]
     }
   }
 
+  "fetchPeersProposals" - {
+    "should update peersProposals" in {
+      val redownloadService = RedownloadService[IO](dao)
+      val updateNewProps = redownloadService.updatePeerProps(proposals)
+
+      updateNewProps.unsafeRunSync().values.forall(_.size == numFacilitators) shouldBe true
+    }
+
+    "should not update peersProposals if a new proposal at the same height as an old proposal is recieved" in {
+      val redownloadService = RedownloadService[IO](dao)
+      val invalidProposdals = proposals :+ (proposals.head._1, RecentSnapshot("invalidProposdals", 0L, Map.empty))
+      val res = redownloadService.updatePeerProps(invalidProposdals).unsafeRunSync()
+
+      res.values.forall(_.size == numFacilitators) shouldBe true
+    }
+
+    "should not update peersProposals with a duplicate proposal" in {
+      val redownloadService = RedownloadService[IO](dao)
+      val invalidProposdals = proposals :+ proposals.head
+      val res = redownloadService.updatePeerProps(invalidProposdals).unsafeRunSync()
+      val check = res.values.forall(_.size == numFacilitators)
+      check shouldBe true
+    }
+  }
+
+  "recalculateMajoritySnapshot" - {
+    "should return a majority snapshot when 50% majority achieved" in {
```

Comment:
`should return a majority snapshot when ...` - shouldn't it be a majority state chooser unit tests?

Reply:
This and below does have similar tests in the majority state chooser test. We really should keep these since there'e extra logic in recalculateMajoritySnapshot that forms the data used by the pure function in majorityStateChooser

Reply:
If it's tested in majority state chooser, there is no point to duplicate those tests. If it does some extra logic, most likely it should be inside majority state chooser too. Here you should mock the majority state chooser answers and check only the logic of redownload service